sdk/core/System.ClientModel/src/Pipeline/ClientRetryPolicy.cs

```diff
@@ -0,0 +1,238 @@
+                message.Response?.ContentStream?.Dispose();
+
+                message.RetryCount++;
+                OnTryComplete(message);
+
+                continue;
+            }
+
+            if (thisTryException != null)
+            {
+                // Rethrow if there's only one exception.
+                if (allTryExceptions!.Count == 1)
+                {
+                    ExceptionDispatchInfo.Capture(thisTryException).Throw();
+                }
+
+                throw new AggregateException($"Retry failed after {message.RetryCount + 1} tries.", allTryExceptions);
+            }
+
+            // ShouldRetry returned false this iteration and the last request
+            // we sent didn't cause an exception to be thrown.
+            // So, we're done.  Exit the while loop.
+            break;
+        }
+    }
+
+    protected virtual void OnSendingRequest(PipelineMessage message) { }
+
+    protected virtual ValueTask OnSendingRequestAsync(PipelineMessage message) => default;
+
+    protected virtual void OnRequestSent(PipelineMessage message) { }
+
+    protected virtual ValueTask OnRequestSentAsync(PipelineMessage message) => default;
+
+    protected virtual void OnTryComplete(PipelineMessage message) { }
+
+    internal bool ShouldRetry(PipelineMessage message, Exception? exception)
+    {
+        // If there was no exception and we got a success response, don't retry.
+        if (exception is null && message.Response is not null && !message.Response.IsError)
+        {
+            return false;
+        }
+
+        return ShouldRetryCore(message, exception);
+    }
+
+    internal async ValueTask<bool> ShouldRetryAsync(PipelineMessage message, Exception? exception)
+    {
+        // If there was no exception and we got a success response, don't retry.
+        if (exception is null && message.Response is not null && !message.Response.IsError)
+        {
+            return false;
+        }
+
+        return await ShouldRetryCoreAsync(message, exception).ConfigureAwait(false);
+    }
+
+    protected virtual bool ShouldRetryCore(PipelineMessage message, Exception? exception)
+    {
+        if (message.RetryCount >= _maxRetries)
+        {
+            // We've exceeded the maximum number of retries, so don't retry.
+            return false;
+        }
+
+        return exception is null ?
+            IsRetriable(message) :
+            IsRetriable(message, exception);
+    }
+
+    protected virtual ValueTask<bool> ShouldRetryCoreAsync(PipelineMessage message, Exception? exception)
+        => new(ShouldRetryCore(message, exception));
+
+    internal TimeSpan GetNextDelay(PipelineMessage message, int tryCount)
+    => GetNextDelayCore(message, tryCount);
+
+    protected virtual TimeSpan GetNextDelayCore(PipelineMessage message, int tryCount)
+    {
+        // Default implementation is exponential backoff
+        return TimeSpan.FromMilliseconds((1 << (tryCount - 1)) * _initialDelay.TotalMilliseconds);
```

Comment:
We need to apply a jitter - https://github.com/Azure/azure-sdk-for-net/blob/main/sdk/core/Azure.Core/src/DelayStrategy.cs#L84-L93

Reply:
Cool, I'll add it in.  I had wanted to simplify but if it's required and we don't expose the configuration value for now, I don't think it adds unnecessary complexity.  Thanks for flagging this!

Reply:
Addressed

Reply:
Actually, @JoshLove-msft, it looks like adding jitter into the ClientModel retry policy has broken Azure.Core tests in the feature branch where I'm integrating Azure.Core and System.ClientModel.  So, two questions here:
1. Can you help me understand the jitter requirement and whether this is a hard requirement or a "value add" that we could postpone to a later release?
2. It may be that I can find a better design that enables adding it without regressing Azure.Core functionality -- if it's required, I'll add an issue to track adding that in so it doesn't block this PR.

Thanks!

Reply:
Tracking with https://github.com/Azure/azure-sdk-for-net/issues/41393

Reply:
I assume it broke the tests because the tests were assuming that there were deterministic delays. I don't think any functionality has been regressed. I think having jitter is probably a requirement given the trouble customers can get into without having jitter applied. Here is a [good article](https://aws.amazon.com/builders-library/timeouts-retries-and-backoff-with-jitter/) on the importance of jitter.

Reply:
Perhaps!  Per @christothes's comment in a Core team discussion (I think it was him...), would it make sense to change the tests separately from the change to the behavior?  If I were able to share a list of the tests in question, @JoshLove-msft, do you think you'd be able to make them appropriately tolerant to additional jitter?  Then I could pull those into the Azure.Core 2.0 feature branch and see if I can add the jitter in a non-breaking way.  Let me know if you're up for that.  In the meantime I have an issue tracking adding the jitter in that we can prioritize however is needed.